src/core/signals.cpp

```diff
@@ -22,17 +23,17 @@ namespace Core {
 static LONG WINAPI SignalHandler(EXCEPTION_POINTERS* pExp) noexcept {
     const auto* signals = Signals::Instance();
 
-    auto* code_address = reinterpret_cast<void*>(pExp->ContextRecord->Rip);
+    auto* code_address = Common::GetRip(pExp);
```

Comment:
For this and the other parameters, if we're passing the context into the signal handler anyway we might as well just make it so the only parameter is the context, then the handler function can use these common helpers to get the parts it needs out. For both access violation and illegal instruction.

Reply:
Sure, except the access violation still needs the fault address param as it gets it from info->si_addr on linux